pkg/apis/core/v1beta1/types_cloudprofile.go

```diff
@@ -297,6 +311,29 @@ type Limits struct {
 	MaxNodesTotal *int32 `json:"maxNodesTotal,omitempty" protobuf:"varint,1,opt,name=maxNodesTotal"`
 }
 
+// CloudProfileStatus contains the status of the cloud profile.
+type CloudProfileStatus struct {
+	// KubernetesVersions contains the statuses of the kubernetes versions.
+	KubernetesVersions []ExpirableVersionStatus `json:"kubernetesVersions,omitempty" protobuf:"bytes,1,name=kubernetesVersions"`
+	// MachineImageVersions contains the statuses of the machine image versions.
+	MachineImageVersions []MachineImageVersionStatus `json:"machineImageVersions,omitempty" protobuf:"bytes,2,name=machineImageVersions"`
+}
+
+type MachineImageVersionStatus struct {
```

Comment:
Requires docstring. Example:
```suggestion
// MachineImageVersionStatus contains the status of a machine image and its versions.
type MachineImageVersionStatus struct {
```